packages/transactions/lib/createTransaction.ts

```diff
@@ -0,0 +1,64 @@
+import db, { TransactionItem, Transaction } from '@play-money/database'
+import { checkUserBalance } from './getUserBalances'
+
+export type TransactionItemInput = Pick<TransactionItem, 'userId' | 'currencyCode' | 'amount'>
+
+export type TransactionInput = Pick<Transaction, 'description' | 'marketId' | 'type' | 'creatorId'> & {
+  transactionItems: Array<TransactionItemInput>
+}
+
+export async function createTransaction({
+  description,
+  marketId,
+  transactionItems,
+  type,
+  creatorId,
+}: TransactionInput) {
+  const currencySums: { [currencyCode: string]: number } = {}
+
+  transactionItems.forEach((item) => {
+    if (!currencySums[item.currencyCode]) {
+      currencySums[item.currencyCode] = 0
+    }
+
+    currencySums[item.currencyCode] += item.amount
+  })
+
+  Object.entries(currencySums).forEach(([currencyCode, sum]) => {
+    if (sum !== 0) {
+      throw new Error(`TransactionItems do not balance for currencyCode ${currencyCode}`)
+    }
+  })
+
+  await Promise.all(
```

Comment:
Does this work with the exchanger? While a market is open, it has a positive primary balance and negative YES and NO balances. I agree that we should have some checks here though, so maybe the exchanger just needs a special case?

Reply:
Yeah likely need to rethink checking actual balances for non-users!

Reply:
For now I switched it to log for debugging purposes, but can likely be removed in favor of checking only user balances